Adding ESMValTool version compatibility to the individual lessons

rswamina opened the pull request:
# Pull Request checklist

We appreciate your time and effort to improve the tutorial. Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution.

* * *

## Before you start

- [x] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
- [x] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss your idea. This allows your contributions to be incorporated into the tutorial.

## Tasks

- [x] Give this pull request a descriptive title.
- [x] If you are contributing to existing lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] If you are making a new episode, please make sure the content conforms to the `Lesson organization` and `Lesson formatting` sections in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [x] Preferably Codacy checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why.
- [x] Preview changes on your machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
- [x] All code instructions have been tested.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes [#240]({https://github.com/ESMValGroup/ESMValTool_Tutorial/issues/240})

LisaBock reviewed (commented):
The schedule on the first page is now really confusing and not well readable. I am wondering if there is a better place to indicate the compatible versions (which is indeed important).

rswamina commented:
> The schedule on the first page is now really confusing and not well readable. I am wondering if there is a better place to indicate the compatible versions (which is indeed important).

Not sure where else to put it so it is seen right away. Any suggestions? I'd be happy to make the change.

remi-kazeroni commented:
> > The schedule on the first page is now really confusing and not well readable. I am wondering if there is a better place to indicate the compatible versions (which is indeed important).
> 
> Not sure where else to put it so it is seen right away. Any suggestions? I'd be happy to make the change.

I agree, the episode title would probably be too long. What about:
- adding the compatible version number in the footer (i.e. [here](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/6a34ade777b5416da7af35ced3e23e4a4f4aed02/_includes/lesson_footer.html#L30) or [here](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/6a34ade777b5416da7af35ced3e23e4a4f4aed02/_includes/lesson_footer.html#L50))?
- adding a third entry below "Teaching: X min, Episode: X min, Compatibility: ESMValTool v.2.5.0" (probably requires to change the template a bit)?

SarahAlidoost commented:
> > > The schedule on the first page is now really confusing and not well readable. I am wondering if there is a better place to indicate the compatible versions (which is indeed important).
> > 
> > 
> > Not sure where else to put it so it is seen right away. Any suggestions? I'd be happy to make the change.
> 
> I agree, the episode title would probably be too long. What about:
> 
> * adding the compatible version number in the footer (i.e. [here](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/6a34ade777b5416da7af35ced3e23e4a4f4aed02/_includes/lesson_footer.html#L30) or [here](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/6a34ade777b5416da7af35ced3e23e4a4f4aed02/_includes/lesson_footer.html#L50))?
> * adding a third entry below "Teaching: X min, Episode: X min, Compatibility: ESMValTool v.2.5.0" (probably requires to change the template a bit)?

I like the second suggestion :+1: 
![image](https://user-images.githubusercontent.com/55081872/176719010-223a4bd5-6701-4e79-819f-187592717b25.png)

Not the template, but it can be implemented in [this file](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/main/_includes/episode_overview.html)  as:

@rswamina  you can add this code after line 29: 
`{% assign compatibility = page.compatibility %}`

and this code after line 53: 
```
      <br/>
      <strong>Compatibility:</strong> {{ compatibility }}
```

and this code to each lesson:
`compatibility: ESMValTool v.2.5.0`

SarahAlidoost commented:
> ![image](https://user-images.githubusercontent.com/55081872/176719010-223a4bd5-6701-4e79-819f-187592717b25.png)
> 

@rswamina @LisaBock @remi-kazeroni shall I make a new pull request to implement this change?

LisaBock commented:
> > ![image](https://user-images.githubusercontent.com/55081872/176719010-223a4bd5-6701-4e79-819f-187592717b25.png)
> 
> @rswamina @LisaBock @remi-kazeroni shall I make a new pull request to implement this change?

Why would you like to open a new pull request? Isn't here the right place?

rswamina commented:
@SarahAlidoost @LisaBock @remi-kazeroni  - I have to complete this based on the last comments on where to put the version number. I assume the current pull request is good ? I guess I am not sure why we need a new pull request.

LisaBock commented:
> @SarahAlidoost @LisaBock @remi-kazeroni - I have to complete this based on the last comments on where to put the version number. I assume the current pull request is good ? I guess I am not sure why we need a new pull request.

@rswamina I think we could change it to Sarah's suggestion:

> > > ![image](https://user-images.githubusercontent.com/55081872/176719010-223a4bd5-6701-4e79-819f-187592717b25.png)

SarahAlidoost commented:
> @SarahAlidoost @LisaBock @remi-kazeroni - I have to complete this based on the last comments on where to put the version number. I assume the current pull request is good ? I guess I am not sure why we need a new pull request.

@rswamina shall we add a new section "compatibility" as shown [here](https://github.com/ESMValGroup/ESMValTool_Tutorial/pull/241#issuecomment-1282268167) instead of adding it to the title of the lessons? Would you like to implement it? or shall I implement it? :thinking: 

@LisaBock I was thinking to create a new pull request because I don't need to undo the changes which are added to the lesson titles in this branch, and also I don't need to fix the merge conflicts. :blush:

rswamina commented:
@SarahAlidoost @LisaBock - I have added the compatibility key to the template and a version for every episode. I am a bit unsure if I have put the correct version of the tool though. If everything looks ok, please merge.

SarahAlidoost commented:
> @SarahAlidoost @LisaBock - I have added the compatibility key to the template and a version for every episode. I am a bit unsure if I have put the correct version of the tool though. If everything looks ok, please merge.

Thanks for the implementation. I think the syntax for specifying the compatibility is not correct. For example, in the lesson "Introduction", we see `Compatibility: ESMValTool v.2.5.0`. This means that this lesson is only compatible with v.2.5.0 version and not other versions. This should be `Compatibility: ESMValTool >=v.2.5.0`. Also, let's remove the last number of the version i.e. `Compatibility: ESMValTool >=v.2.5` is enough. @remi-kazeroni what do you think?

SarahAlidoost commented on _episodes/01-quickstart.md:
```diff
@@ -2,6 +2,8 @@
 teaching: 2
 exercises: 8
+compatibility: ESMValTool v.2.6.0
+
```
Please remove line 35 in this lesson where it is mentioned `> This quickstart guide is compatible with ESMValTool v2.6.0.`

rswamina replied:
Done.

LisaBock commented on _episodes/09-cmorization.md:
```diff
@@ -2,6 +2,7 @@
 title: "CMORization: adding new datasets to ESMValTool"
 teaching: 15
 exercises: 45
+compatibility: ESMValTool v.2.5.0
```
That should be v.2.6.0. Rémi wrote that recently.

rswamina replied:
Ok. I can change this.

rswamina replied:
changed.

LisaBock reviewed (changes requested):
Thanks @rswamina !

For two epsiodes I think we could change it to v2.6.0. Or did I miss required changes?

rswamina commented on _episodes/00-introduction.md:
```diff
@@ -2,6 +2,8 @@
 title: "Introduction"
 teaching: 5
 exercises: 10
+compatibility: ESMValTool v.2.5.0
```
@LisaBock  - I wasn't sure, should it be v2.6? If so, I will change it.

rswamina replied:
Fixed now.

rswamina commented:
@LisaBock @SarahAlidoost and @remi-kazeroni - I think I have addressed all comments so far. If you feel further changes are needed or there are any mistakes to be fixed, please let me know and I will make/fix them. Otherwise at this time I don't have any more to add to this PR.

remi-kazeroni commented on _episodes/01-quickstart.md:
```diff
@@ -29,8 +31,8 @@ keypoints:
-> This quickstart guide is compatible with ESMValTool v2.6.0.
+
+
  {: .callout}
```
I think that would solved the failed test
```suggestion

```

rswamina replied:
Fixed this and the version numbers.

remi-kazeroni reviewed (commented):
Thanks a lot for picking up on that @rswamina! This looks fine to me. I think it's probably fine to only refer to the latest ESMValTool version for which the episode works. It might be counter-productive to tell newcomers that they could do the tutorial with old versions. It's also good way to push us to keep the tutorial up to date.

I think it would be better to refer to exact version numbers with 3 digits (e.g. 2.6.0) but I don't have a strong opinion on this. Also maybe use `v2.6.0` instead of `v.2.6.0` (one dot less)? See version numbers on [Zenodo](https://zenodo.org/search?page=1&size=20&q=conceptrecid:%223401363%22&sort=-version&all_versions=True).

The tests are currently failing and that should be fixed with my code suggestion.

SarahAlidoost reviewed (approved):
@rswamina thanks :+1:

LisaBock reviewed (approved):
Thanks @rswamina !
